src/scroll-timeline-css-parser.js

```diff
@@ -0,0 +1,327 @@
+// 1 - Extracts @scroll-timeline and saves it in scrollTimelineOptions.
+// 2 - If we find any animation-timeline in any of the CSS Rules, 
+// we will save objects in a list named cssRulesWithTimelineName
+export class StyleParser {
+  constructor() {
+    this.cssRulesWithTimelineName = [];
+    this.scrollTimelineOptions = new Map(); // save options by name
+  }
+
+  transpileStyleSheet(sheetSrc, srcUrl) {
+    // AdhocParser
+    const p = {
+      sheetSrc: sheetSrc,
+      index: 0,
+      name: srcUrl,
+    };
+
+    while (p.index < p.sheetSrc.length) {
+      this.eatWhitespace(p);
+      if (p.index >= p.sheetSrc.length) break;
+      if (this.lookAhead("/*", p)) {
+        while (this.lookAhead("/*", p)) {
+          this.eatComment(p);
+          this.eatWhitespace(p);
+        }
+        continue;
+      }
+      if (this.lookAhead("@scroll-timeline", p)) {
+        const { scrollTimeline, startIndex, endIndex } = this.parseScrollTimeline(p);
+        this.scrollTimelineOptions.set(scrollTimeline.name, scrollTimeline);
+      } else {
+        const rule = this.parseQualifiedRule(p);
+        if (!rule) continue;
+        this.handleScrollTimelineProps(rule, p);
+      }
+    }
+
+    // If this sheet has no srcURL (like from a <style> tag), we are
+    // done. Otherwise, we have to find `url()` functions and resolve
+    // relative and path-absolute URLs to absolute URLs.
+    if (!srcUrl) {
+      return p.sheetSrc;
+    }
+
+    p.sheetSrc = p.sheetSrc.replace(
+      /url\(["']*([^)"']+)["']*\)/g,
```

Comment:
Regex tend to be hard to read.  Suggest assigning to a const with a comment to better document what it is being used for.